applications/app/templates/serviceWorker.scala.js

```diff
@@ -139,3 +140,64 @@ this.addEventListener('fetch', function (event) {
     }
 });
 
+self.addEventListener('activate', function(event) {
+});
+
+self.addEventListener('push', function(event) {
+
+    event.waitUntil(
+        self.registration.pushManager.getSubscription().then(function (sub) {
+            var gcmInd = sub.endpoint.substring(sub.endpoint.lastIndexOf('/') + 1);
+
+            var endpoint = '@{JavaScript(Configuration.Notifications.latestMessageUrl)}/' + gcmInd;
+            fetch(endpoint, {
+                method: 'get',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json'
+                }
+            }).then(function (response) {
+               return response.json();
+            })
+            .then(function (json) {
+
+               if(json.status === "ok")
+                    var messages = json.messages;
+
+                    messages.forEach( function(message) {
+                        var data = {topic: message.topic};
+                        self.registration.showNotification(message.title, {
+                            body: message.body,
+                            icon: '@{JavaScript(Static("images/favicons/114x114.png").path)}',
+                            tag: message.title,
+                            data: data
+                        });
+                    });
+                })
+
+        })
+    );
+});
+
+self.addEventListener('notificationclick', function(event){
+
+        event.notification.close();
+        var url = '@{JavaScript(Configuration.javascript.pageData.get("guardian.page.host").getOrElse("https://www.theguardian.com"))}/' + event.notification.data.topic + "?CMP=not_b-webalert";
```

Comment:
This: `Configuration.javascript.pageData.get("guardian.page.host")`
could become: `Configuration.site.host`
I think?